### ISSUE TITLE
`java.lang.NoSuchMethodError` on Android 6.0.1 (Galaxy Grand Prime Plus)

### ISSUE THREAD

**Issue by MohammadEhsanMohit** [gojek/courier-android#87]
**Description**:
I am encountering a `java.lang.NoSuchMethodError` in my Android application when using the Courier library on a device running Android 6.0.1 (Galaxy Grand Prime Plus). The error seems related to the `isDefault()` method in the `java.lang.reflect.Method` class, which is not available in this Android version.

**Error Stack Trace**:
```
FATAL EXCEPTION: DefaultDispatcher-worker-10
Process: com.example.app, PID: 13638
java.lang.NoSuchMethodError: No virtual method isDefault()Z in class Ljava/lang/reflect/Method; or its super classes (declaration of 'java.lang.reflect.Method' appears in /system/framework/core-libart.jar)
	at com.gojek.courier.utils.d.a(SourceFile:1)
	at com.gojek.courier.utils.RuntimePlatform$Java8.isDefaultMethod(SourceFile:6)
	at com.gojek.courier.stub.StubInterface$Factory.findStubMethods(SourceFile:28)
	at com.gojek.courier.stub.StubInterface$Factory.create(SourceFile:9)
	at com.gojek.courier.Courier.create(SourceFile:1)
	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(SourceFile:4)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(SourceFile:15)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(SourceFile:29)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(SourceFile:1)
```

**Steps to Reproduce**:
1. Use the Courier library in an Android application.
2. Test the app on a device running Android 6.0.1 (Galaxy Grand Prime Plus).
3. Observe the crash when trying to establish a connection using the Courier library.

**Expected Behavior**:
The Courier library should handle compatibility with older Android versions that do not have the `isDefault()` method.

**Additional Information**:
- This issue does not occur on devices with Android versions 7.0 and above.
- The issue appears to stem from compatibility with the `java.lang.reflect.Method.isDefault()` method, which is available in newer Android versions.

Would it be possible to include a check to handle older versions of Android that do not support this method, or suggest any configuration changes on my end to avoid this issue?

Thank you for your support and for maintaining this library.

**Comment by sumitjat3** [comment]
Hey @MohammadEhsanMohit  Thanks for detailed explanation regarding issue, I tried to reproduce this issue in API 21 which has minfiyEnabled true but not able to reproduce it.

Can you try to reproduce this in [Example Module](https://github.com/gojek/courier-android/blob/main/app/build.gradle)?

**Comment by MohammadEhsanMohit** [comment]
> Hey @MohammadEhsanMohit Thanks for detailed explanation regarding issue, I tried to reproduce this issue in API 21 which has minfiyEnabled true but not able to reproduce it.
> 
> Can you try to reproduce this in [Example Module](https://github.com/gojek/courier-android/blob/main/app/build.gradle)?

Hey @sumitjat3, you're welcome! As I mentioned before, desugaring needs to be enabled in your project. The reason is that desugaring helps make certain Java features available on older Android versions by replacing or adapting missing classes. For example, it changes java.util.Optional to j$.util.Optional.

To enable desugaring for Java 8+ APIs on older Android versions in your project, you’ll need to add support for core library desugaring in the dependencies section and configure it in the compileOptions section. as example:

```gradel
android {
    compileSdk  ..
    ...
    compileOptions {
        sourceCompatibility JavaVersion.VERSION_11
        targetCompatibility JavaVersion.VERSION_11
        // Enable core library desugaring
        coreLibraryDesugaringEnabled true
    }

....
}

dependencies {
    ...
    // Core library desugaring support
    coreLibraryDesugaring 'com.android.tools:desugar_jdk_libs:2.0.3'
   .....
}

```

Also, try the release version to reproduce the problem. Sorry, but when I tried to release the sample project you asked for, I encountered an error stating that Kotlin libraries are not allowed to be minified. If you’re able to build a minified release, please enable desugaring and test it again!

Thank you @sumitjat3 !

**Comment by sumitjat3** [comment]
Hey @MohammadEhsanMohit  Still not able to reproduce it, for now what we can do is maybe you can try the local version  and try if [fix raised](https://github.com/gojek/courier-android/pull/89) by you looks good or not.

if looks good we are good to merge for this.

**Comment by deepanshu42** [comment]
Fix is available in [0.3.1](https://github.com/gojek/courier-android/releases/tag/0.3.1)
